### PR TITLE
CardChangedType: use Enum Flags for removeAllSubTypes

### LINKED ISSUE #3223: CardChangedType removeType params into EnumSetFlags?
Imo Changed Card Types has way too many parameters:

```
public CardChangedType(final CardType addType0, final CardType removeType0, final boolean addAllCreatureTypes0,
            final boolean removeSuperType0, final boolean removeCardType0, final boolean removeSubType0,
            final boolean removeLandType0, final boolean removeCreatureType0, final boolean removeArtifactType0, final boolean removeEnchantmentTypes0) {
```

i want to refactor them into an EnumSet Bitflag:
```
public CardChangedType(final CardType addType0, final CardType removeType0, final boolean addAllCreatureTypes0,
            final EnumSet<FlagType> removeFlagTypes0) {
```

### PR DESCRIPTION
**Pull request by Hanmac**
Closes #3223 

Maybe First part or first draft?

Currently it is just Internal Flags without Changing the Card Scripts

More Opinions about this?

- [x] Should the Enum be its own file, ~~or is `CardChangedType.Remove` okay~~?

Should we update the `RemoveLandTypes` params to be something like `RemoveAllTypes$ Land` ?

### CODE REVIEW

File: forge-core/src/main/java/forge/card/CardChangedType.java [hunk 1]
```diff
@@ -26,31 +28,30 @@
  * @version $Id: CardType.java 24393 2014-01-21 06:27:36Z Max mtg $
  */
 public class CardChangedType {
+
+    public enum Remove {
```
Comment (tehdiplomat) [inline forge-core/src/main/java/forge/card/CardChangedType.java:32]: If this is public, it probably should be a different file.

Comment (Agetian) [reply forge-core/src/main/java/forge/card/CardChangedType.java:32]: :+1: Agreed

**Comment by Hanmac** [comment]
@tehdiplomat @Agetian @tool4ever something more to say about this MR?

**Review by tool4ever — APPROVED** [review]
Seems fine

**Review by Agetian — APPROVED** [review]

**Pull request merged**